src/software/thunderscope/robot_diagnostics/drive_and_dribbler_widget.py

```diff
@@ -16,32 +24,56 @@ def __init__(self, proto_unix_io: ProtoUnixIO) -> None:
         self.input_a = time.time()
         self.constants = tbots_cpp.create2021RobotConstants()
         QWidget.__init__(self)
+
         layout = QVBoxLayout()
+        self.control_widget = QStackedWidget()
 
         self.proto_unix_io = proto_unix_io
 
-        # Add widgets to layout
-        layout.addWidget(self.setup_direct_velocity("Drive"))
-        layout.addWidget(self.setup_dribbler("Dribbler"))
+        # create swappable widget system using stacked widgets
+        self.direct_velocity_widget = self.setup_direct_velocity(
+            "Control - Direct Velocity"
+        )
+        self.per_motor_widget = self.setup_per_motor("Control - Per Motor")
+        self.control_widget.addWidget(self.direct_velocity_widget)
+        self.control_widget.addWidget(self.per_motor_widget)
 
-        self.enabled = True
+        layout.addWidget(self.setup_drive_switch_radio("Drive Mode Switch"))
+        layout.addWidget(self.control_widget)
+        layout.addWidget(self.setup_dribbler("Dribbler"))
 
+        # default to direct velocity
+        self.use_direct_velocity.click()
         self.setLayout(layout)
+        self.toggle_dribbler_sliders(True)
 
     def refresh(self) -> None:
-        """Refresh the widget and send the a MotorControl message with the current values"""
+        """Refresh the widget and send the MotorControl message with the current values depending on the ControlMode"""
         motor_control = MotorControl()
         motor_control.dribbler_speed_rpm = int(self.dribbler_speed_rpm_slider.value())
-
-        motor_control.direct_velocity_control.velocity.x_component_meters = (
-            self.x_velocity_slider.value()
-        )
-        motor_control.direct_velocity_control.velocity.y_component_meters = (
-            self.y_velocity_slider.value()
-        )
-        motor_control.direct_velocity_control.angular_velocity.radians_per_second = (
-            self.angular_velocity_slider.value()
-        )
+        if self.control_mode == ControlMode.MOTOR:
```

Comment:
this refresh looks flipped. It looks like `ControlMode.VELOCITY` corresponds to this `if` conditional code path